Assets/Scripts/Model/Content/SecondEdition/Pilots/NabooRoyalN1Starfighter/NabooHandmaiden.cs

```diff
@@ -0,0 +1,285 @@
+
+            return result;
+        }        
+    }
+}
+
+namespace Conditions
+{
+    public class Decoyed : GenericToken
+    {
+        public GenericUpgrade SourceUpgrade;
+
+        public Decoyed(GenericShip host) : base(host)
+        {
+            Name = ImageName = "Decoyed Condition";
+            Temporary = false;
+            // Tooltip = "https://raw.githubusercontent.com/Sandrem/xwing-data2-test/master/images/conditions/decoyed.png";
+            Tooltip = "https://images-cdn.fantasyflightgames.com/filer_public/7e/38/7e38aca8-b0ea-4ddc-8ec4-64efca1544c8/swz40_decoyed.png";
+        }
+
+        public override void WhenAssigned()
+        {
+            SubscribeToDecoyedConditionEffects();
+        }
+
+        public override void WhenRemoved()
+        {
+            UnsubscribeFromDecoyedConditionEffects();
+        }
+
+        private void SubscribeToDecoyedConditionEffects()
+        {
+            // Remove any other Decoy conditions on friendly ships
+            Host.OnGenerateDiceModifications += AddDecoyedResultModification;
+        }
+
+        private void UnsubscribeFromDecoyedConditionEffects()
+        {
+            Host.OnGenerateDiceModifications -= AddDecoyedResultModification;
+        }
+
+        private void AddDecoyedResultModification(GenericShip ship)
+        {
+            DecoyedAction action = new DecoyedAction()
+            {
+                HostShip = Host,
+                SourceUpgrade = SourceUpgrade
+            };
+            Host.AddAvailableDiceModification(action);    
+        }
+    }
+}
+
+namespace ActionsList
+{
+    public class DecoyedAction : GenericAction
+    {
+        public GenericUpgrade SourceUpgrade;
+
+        public DecoyedAction()
+        {
+            Name = DiceModificationName = "Decoyed";
+        }
+
+        public override bool IsDiceModificationAvailable()
+        {
+            bool result = true;
+
+            // mod only works on defense
+            if (HostShip != Combat.Defender || Combat.AttackStep != CombatStep.Defence) result = false;
+
+            // handmaidens in attack arc with evades to spend
+            int handmaidens = Roster.AllShips.Count(v =>
+                v.Value.PilotInfo.PilotName == "Naboo Handmaiden"
+                && new ShotInfo(Combat.Attacker, v.Value, Combat.Attacker.PrimaryWeapons).InArc
+                && v.Value.Tokens.HasToken(typeof(EvadeToken))
+            );
+            if (handmaidens == 0) result = false;
+            Messages.ShowInfo("Handmaidens in Range: " + handmaidens);
+            return result;
+        }
+
+        private class OptimizedPrototypeDecisionSubPhase : DecisionSubPhase { }
+
+        public override void ActionEffect(System.Action callBack)
+        {
+            var newSubPhase = Phases.StartTemporarySubPhaseNew<OptimizedPrototypeDecisionSubPhase>(Name, callBack);
```

Comment:
I'd love for some advice on the best way to handle registering actions for each Handmaiden in arc of the attack. I've copied this pattern from Director Krennic but am wondering if implementation a la selfless would be better.